bundles/org.openhab.binding.saicismart/src/main/java/org/openhab/binding/saicismart/internal/SAICiSMARTHandler.java

```diff
@@ -0,0 +1,278 @@
+        super(thing);
+        this.timeZoneProvider = timeZoneProvider;
+        lastAlarmMessage = ZonedDateTime.now(getTimeZone());
+        lastCarActivity = ZonedDateTime.now(getTimeZone());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(SAICiSMARTBindingConstants.CHANNEL_FORCE_REFRESH) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_FORCE_REFRESH, OnOffType.from(false));
+            // update internal activity date, to query the car for about a minute
+            notifyCarActivity(ZonedDateTime.now(getTimeZone()).minus(9, ChronoUnit.MINUTES), true);
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.ON) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.ON);
+            // enable air conditioning
+            try {
+                sendACCommand((byte) 5, (byte) 8);
+            } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                logger.warn("A/C On Command failed", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_SWITCH_AC) && command == OnOffType.OFF) {
+            // reset channel to off
+            updateState(CHANNEL_SWITCH_AC, OnOffType.OFF);
+            // disable air conditioning
+            try {
+                sendACCommand((byte) 0, (byte) 0);
+            } catch (URISyntaxException | ExecutionException | TimeoutException | InterruptedException e) {
+                logger.warn("A/C Off Command failed", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_LAST_ACTIVITY) && command instanceof DateTimeType) {
+            // update internal activity date from external date
+            notifyCarActivity(((DateTimeType) command).getZonedDateTime(), true);
+        }
+    }
+
+    protected @Nullable SAICiSMARTBridgeHandler getBridgeHandler() {
+        return (SAICiSMARTBridgeHandler) super.getBridge().getHandler();
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SAICiSMARTVehicleConfiguration.class);
+
+        ThingBuilder thingBuilder = editThing();
+        updateThing(thingBuilder.build());
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Validate configuration
+        if (this.config.vin.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "VIN is empty!");
+            return;
+        }
+
+        // just started, make sure we start querying
+        notifyCarActivity(ZonedDateTime.now(getTimeZone()), true);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::updateStatus, 2,
+                SAICiSMARTBindingConstants.REFRESH_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    private void updateStatus() {
+        if (lastCarActivity.isAfter(ZonedDateTime.now().minus(10, ChronoUnit.MINUTES))) {
```

Comment:
Can you make 10 minutes a class level constant?

Reply:
SAICiSMARTBindingConstants.REFRESH_INTERVAL

Reply:
and SAICiSMARTBindingConstants.POLLING_ACTIVE_MINS